src/ImGui/Desktop/Fusee.ImGui.Desktop/Templates/ImGuiFolderPicker.cs

```diff
@@ -0,0 +1,372 @@
+                ImGui.EndTooltip();
+                ImGui.PopStyleVar();
+            }
+
+            // Folder Browser
+            ImGui.NewLine();
+            ImGui.PushStyleColor(ImGuiCol.ChildBg, FileSelectionMenuBackground.ToUintColor());
+            ImGui.PushStyleVar(ImGuiStyleVar.WindowPadding, new Vector2(10, 10));
+
+            ImGui.BeginChild($"DriveSelection##{_folderPickerCount}", new Vector2(DriveSelectionWidth, BrowserHeight), false, ImGuiWindowFlags.AlwaysUseWindowPadding | ImGuiWindowFlags.AlwaysAutoResize);
+            // Drive Selection
+            var driveCount = 0;
+            foreach (var drive in DriveInfo.GetDrives())
+            {
+                if (drive.IsReady)
+                {
+                    if (ImGui.Selectable($"{drive.Name} {drive.DriveType}##{_folderPickerCount}"))
+                    {
+                        RootFolder = new DirectoryInfo(drive.Name);
+                        LastOpenendFolders.Push(CurrentOpenFolder);
+                        CurrentOpenFolder = new DirectoryInfo(drive.Name);
+                    }
+                    driveCount++;
+                }
+            }
+            ImGui.EndChild();
+            ImGui.SameLine();
+
+            if (ImGui.BeginChild($"#FolderBrowser##{_folderPickerCount}", new Vector2(FolderTextInputWidth, BrowserHeight), false, ImGuiWindowFlags.AlwaysUseWindowPadding | ImGuiWindowFlags.AlwaysAutoResize | ImGuiWindowFlags.HorizontalScrollbar))
+            {
+
+                var fileSystemEntries = GetFileSystemEntries(CurrentOpenFolder.FullName);
+                foreach (var fse in fileSystemEntries)
+                {
+                    var name = fse.Name;
+
+                    ImGui.PushStyleColor(ImGuiCol.Text, FolderColor.ToUintColor());
+                    ImGui.PushStyleColor(ImGuiCol.Header, SelectedColor.ToUintColor());
+
+                    if (ImGui.Selectable(name + "/", CurrentlySelectedFolder?.Name == name, ImGuiSelectableFlags.DontClosePopups | ImGuiSelectableFlags.AllowDoubleClick))
+                    {
+                        CurrentlySelectedFolder = fse;
+                        if (ImGui.IsMouseDoubleClicked(0))
+                        {
+                            if (ImGui.GetIO().WantCaptureMouse)
+                            {
+                                LastOpenendFolders.Push(CurrentOpenFolder);
+                                CurrentOpenFolder = fse;
+                            }
+                        }
+                    }
+
+                    ImGui.PopStyleColor();
+                    ImGui.PopStyleColor();
+                }
+
+
+                ImGui.PopStyleColor();
+                ImGui.PopStyleVar();
+                ImGui.EndChild();
+            }
+
+            //// Folder Selector
+            ImGui.NewLine();
+            ImGui.BeginChild($"FileSelector##{_folderPickerCount}", new Vector2(-1, -1), false, ImGuiWindowFlags.AlwaysAutoResize);
```

Comment:
Better to use `"FolderSelector"` to avoid confusion...

Reply:
Thanks. Copy-Pasta error.